blog_post_collaboration/deploy_Kedro_pipelines_on_EMR.md

```diff
@@ -0,0 +1,165 @@
+
+Run the Dockerfile  using the following command. This will generate a `pyspark_deps.tar.gz`  file at the  `<output-path>`  specified in the command above. 
+
+```bash
+DOCKER_BUILDKIT=1 docker build --output . <output-path> 
+```
+
+Use this command if your Dockerfile has a different name 
+
+```bash
+DOCKER_BUILDKIT=1 docker build -f Dockerfile-emr-venv --output . <output-path>
+```
+
+## 2. Set up `CONF_ROOT`
+
+By default, Kedro looks at the root `conf` folder for all its configurations (catalog, parameters, globals, credentials, logging) to run the pipelines. However, [this can be customized](https://docs.kedro.org/en/stable/kedro_project_setup/configuration.html#configuration-root) by changing `CONF_ROOT`  in `settings.py`. 
+
+As `conf` directory is not packaged together with `kedro package`, the `conf` directory needs to be available to Kedro separately and that location can be controlled by setting `CONF_ROOT`. 
+
+### For Kedro versions < 0.18.5
+
+-   Change `CONF_ROOT in settings.py` to the location where `conf` directory will be deployed. It could be anything. e.g. `./conf`  or `/mnt1/kedro/conf`. 
+
+### For Kedro versions >= 0.18.5
+
+-   Use the `--conf-source` CLI parameter directly with `kedro run` to specify the path directly. `CONF_ROOT` need not be changed in `settings.py`.
+
+## 3. Package the Kedro project
+
+Package the project using `kedro package`  command. This will create a `.egg` in the `dist`  folder. 
+
+This will be used when doing `spark-submit` to the EMR cluster for specifying the `--py-files` to refer to the source code. 
+
+```bash
+# run at the root directory 
+kedro package
+```
+
+## 4. Create `.tar` for `conf`
+
+The `kedro package` command only packs the source code and yet the `conf/` directory is essential for running any Kedro project.
+
+Therefore it needs to be deployed separately as a `tar.gz`  file. It is important to note that the contents inside the `conf` directory is
+to be zipped and not the `conf`  folder entirely. Use the following command to zip the contents inside the conf directory. This will generate a `conf.tar.gz` file containing all the `catalog.yml`, `parameters.yml` and others for running the Kedro pipeline. It will be used with `spark-submit` for the `--archives` option for unpacking the contents into a `conf` directory .
+
+```bash
+tar -czvf conf.tar.gz --exclude="local" conf/*
+```
+
+## 5. Create entrypoint For Spark application 
+
+Create an `entrypoint.py` file that the Spark Application will use to start the job. 
+
+Example below:
+
+> Note: This file can be modified to take arguments and can be run only
+using `main(sys.argv)` after removing the `params` array. 
+
+Example:
+
+```bash
+python entrypoint.py --pipeline my_new_pipeline --params run_date:2023-02-05,runtime:cloud
+```
+This would mimic the exact `kedro run` behaviour.
+
+```python
+import sys 
+from proj_name.__main__ import main: 
+
+if __name__ == "__main__":
+# params = [ 
```

Comment:
I don't know if this comment style is intuitive

Reply:
I'll turn it into a triple quote with appropriate indentation and add notes for better understanding.